src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -1285,7 +1285,8 @@ private ImmutableArray<ParameterSymbol> ComputeParameters(Binder binder, BasePro
                 }
                 else if ((object)_setMethod != null && param.Name == ParameterSymbol.ValueParameterName)
                 {
-                    diagnostics.Add(ErrorCode.ERR_DuplicateGeneratedName, _location, param.Name);
+                    // note: there can only ever be one location for param, as it cannot be used in partial for a property
+                    diagnostics.Add(ErrorCode.ERR_DuplicateGeneratedName, param.Locations[0], param.Name);
```

Comment:
There are only two call sites for SourcePropertySymbol constructor, both seem to pass non-null location.